### PR TITLE
EVG-6817 add delete button for commit queue patch page

### PR DESCRIPTION
**Pull request by ablack12**
This is so users don't have to wait for the patch to be scheduled to delete an item from the queue from the UI.

Screen Shot 2020-01-06 at 1 36 01 PM

This is the link we reroute to if the deletion works. Alternatively we could reroute to the patch page or something and put up a banner? I think this is clear enough though.

Screen Shot 2020-01-06 at 1 36 27 PM

### CODE REVIEW

File: public/static/js/patch_new.js [hunk 1]
```diff
@@ -94,6 +94,18 @@ mciModule.controller('PatchController', function($scope, $filter, $window, notif
     }
   }
 
+  $scope.deleteFromQueue = function() {
+    var successHandler =  function(resp) {
+        window.location.reload();
+        notificationService.pushNotification("Item successfully cleared from queue", 'success');
```
Comment (ablack12) [inline public/static/js/patch_new.js]: This banner doesn't show up. Is there something about reloading that does weird things?

**Comment by ablack12** [comment]
This is a side question @ybrill but the UI reports position with as zero indexed and the CLI doesn't, meaning this example commit was reported as position 2 from the CLI and position 1 from the UI. I'd like to change the CLI to also report the position as zero indexed. Do you have a preference or a concern?

File: public/static/js/patch_new.js (hunk 1 above)
Comment (ybrill) [reply public/static/js/patch_new.js]: My guess would be that reloading stops execution where it is so this line never gets executed.

**Comment by ybrill** [comment]
@ablack12 I thought this was addressed in #2551. Is this a regression?

File: rest/route/commit_queue.go [hunk 2]
```diff
@@ -123,8 +123,8 @@ func (cq *commitQueueDeleteItemHandler) Run(ctx context.Context) gimlet.Responde
 	}
 
 	response := gimlet.NewJSONResponse(struct{}{})
-	if err := response.SetStatus(http.StatusNoContent); err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "Cannot set HTTP status code to %d", http.StatusNoContent))
+	if err := response.SetStatus(http.StatusOK); err != nil {
```
Comment (ybrill) [inline rest/route/commit_queue.go]: From [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE#Responses) it looks like we'd want to return a 204.

File: public/static/js/patch_new.js (hunk 1 above)
Comment (ablack12) [reply public/static/js/patch_new.js]: pushFlash is a go function for the ui server though and this is a v2 route, so can this not be done? It's not strictly necessary, I just thought it'd be nice.

File: rest/route/commit_queue.go (hunk 2 above)
Comment (ablack12) [reply rest/route/commit_queue.go]: Oh sorry that's fine, I was just doing things with the route and was confused by the response but sounds good.

**Comment by ablack12** [comment]
Looks like I missed Enqueue at the time? Not sure how that slipped through for so long but I pushed a fix.

**Pull request merged**